### ISSUE TITLE
Extreme data duplication on TCP inputs

### ISSUE THREAD

**Issue by joell** [fluent/fluent-bit#342]
I've stumbled across a scenario in which Fluent Bit will duplicate the input it receives via a TCP input.  This seems to be exacerbated by reducing the flush timeout.  This behaviour was observed on the TD Agent Bit v0.11.15 RPMs.

Here is the simplest example I could reduce the problem to:

```sh
# start a simple fluent-bit instance with a short flush timeout
$ td-agent-bit -f1 -i tcp -o stdout &

# quickly pump a bunch of JSON objects into it, forcing output line buffering
$ seq 1 24 | stdbuf -oL sed 's/.*/{"num": &}/' > /dev/tcp/localhost/5170

# output on the console from Fluent Bit
[0] tcp.0: [1501701347, {"num"=>1}]
[1] tcp.0: [1501701347, {"num"=>2}]
[2] tcp.0: [1501701347, {"num"=>3}]
[3] tcp.0: [1501701347, {"num"=>4}]
[4] tcp.0: [1501701347, {"num"=>5}]
[5] tcp.0: [1501701347, {"num"=>6}]
[6] tcp.0: [1501701347, {"num"=>7}]
[7] tcp.0: [1501701347, {"num"=>2}]
[8] tcp.0: [1501701347, {"num"=>3}]
[9] tcp.0: [1501701347, {"num"=>4}]
[10] tcp.0: [1501701347, {"num"=>5}]
[11] tcp.0: [1501701347, {"num"=>6}]
[12] tcp.0: [1501701347, {"num"=>7}]
[13] tcp.0: [1501701347, {"num"=>8}]
[14] tcp.0: [1501701347, {"num"=>9}]
[15] tcp.0: [1501701347, {"num"=>10}]
[16] tcp.0: [1501701347, {"num"=>11}]
[17] tcp.0: [1501701347, {"num"=>12}]
[18] tcp.0: [1501701347, {"num"=>13}]
[19] tcp.0: [1501701347, {"num"=>14}]
[20] tcp.0: [1501701347, {"num"=>15}]
[21] tcp.0: [1501701347, {"num"=>16}]
[22] tcp.0: [1501701347, {"num"=>17}]
[23] tcp.0: [1501701347, {"num"=>18}]
[24] tcp.0: [1501701347, {"num"=>19}]
[25] tcp.0: [1501701347, {"num"=>20}]
[26] tcp.0: [1501701347, {"num"=>21}]
[27] tcp.0: [1501701347, {"num"=>22}]
[28] tcp.0: [1501701347, {"num"=>3}]
[29] tcp.0: [1501701347, {"num"=>4}]
[30] tcp.0: [1501701347, {"num"=>5}]
[31] tcp.0: [1501701347, {"num"=>6}]
[32] tcp.0: [1501701347, {"num"=>7}]
[33] tcp.0: [1501701347, {"num"=>8}]
[34] tcp.0: [1501701347, {"num"=>9}]
[35] tcp.0: [1501701347, {"num"=>10}]
[36] tcp.0: [1501701347, {"num"=>11}]
[37] tcp.0: [1501701347, {"num"=>12}]
[38] tcp.0: [1501701347, {"num"=>13}]
[39] tcp.0: [1501701347, {"num"=>14}]
[40] tcp.0: [1501701347, {"num"=>15}]
[41] tcp.0: [1501701347, {"num"=>16}]
[42] tcp.0: [1501701347, {"num"=>17}]
[43] tcp.0: [1501701347, {"num"=>18}]
[44] tcp.0: [1501701347, {"num"=>19}]
[45] tcp.0: [1501701347, {"num"=>20}]
[46] tcp.0: [1501701347, {"num"=>21}]
[47] tcp.0: [1501701347, {"num"=>22}]
[48] tcp.0: [1501701347, {"num"=>23}]
[49] tcp.0: [1501701347, {"num"=>24}]
```

In this case we have a duplicated output of just over 2x.  However, the duplication gets worse the more data is passed into Fluent Bit.  I have observed duplication ratios of 65x under higher load.

Also, while it is most easily observable at `-f1`, you can still witness the effect at the default flush interval of 5 seconds with a larger input (say, 100 instead of 24 JSON lines).

**Comment by edsiper** [comment]
thanks for reporting this problem. I was able to reproduce it.

note: I didn't knew this trick -> /dev/tcp/localhost/5170 :)

**Comment by edsiper** [comment]
@joell FYI: I've pushed the fixes on git master and they will be backported to 0.11 (released on 0.11.16).

**Comment by edsiper** [comment]
Fixed.
